airflow/models/taskinstance.py

```diff
@@ -885,6 +887,15 @@ def _run_raw_task(
                 context = self.get_template_context()
 
                 task_copy = copy.copy(task)
+
+                # Sensors in `poke` mode can block execution of DAGs when running
+                # with single process executor, thus we change the mode to`reschedule`
+                # to allow parallel task being scheduled and executed
+                if issubclass(type(task_copy), BaseSensorOperator) and \
```

Comment:
Can we not simplify this with following:
```suggestion
                if isintance(task_copy, BaseSensorOperator) and \
```

Reply:
True, good catch!